### ISSUE TITLE
Review read-only styling for Radio Button and Checkbox

### ISSUE THREAD

**Issue by liamsms** [jpmorganchase/salt-ds#3941]
### Area

UI Components

### The problem

When the theme mode is set to dark, selected read-only Radios are difficult to see against a dark background. 

https://storybook.saltdesignsystem.com/?path=/story/core-radio-button--readonly&globals=mode:dark

### The solution

Change the selected read-only foreground color to white when in dark mode.

### Alternatives and examples

.

### Are you a JPMorgan Chase & Co. employee?

- [X] I am an employee of JPMorgan Chase & Co.

**Comment by origami-z** [comment]
Thanks for raising. Will take a look next sprint.

**Comment by joshwooding** [comment]
Should be a stretch goal, needs a more holistic solution. Looking at the read-only styles for checkbox and radio and ensuring that they are okay from a11y pov.

**Comment by mark-tate** [comment]
Bumped to Frappe as non-trivial change

**Comment by mark-tate** [comment]
Frappe Goal: needs discussion with a11y SME , by EOS have a proposal

**Comment by bhoppers2008** [comment]
Update: Described issue to @jake-costa and will review later 24 Sep.

**Comment by mark-tate** [comment]
@bhoppers2008 can we close this in Frappe

**Comment by mark-tate** [comment]
Latte goal: define spec for read-only states

**Comment by jake-costa** [comment]
@bhoppers2008 Can you share a link to the specs for this?

**Comment by origami-z** [comment]
checkbox: https://www.figma.com/design/hWEpxUvJlNV6xy9pVDgxGT/Checkbox?node-id=1209-7099&t=33mCDaztjSB1uAHt-1
radio: https://www.figma.com/design/cxgSJLDjLvVjYuykX4iiIK/Radio-Button?node-id=1332-19408&t=j1lqqxdl2HOSLSyC-1

**Comment by jake-costa** [comment]
Thanks @origami-z, However it looks like the circle and square are both are still falling below 3:1 in read-only state. Read-only elements are not exempt from the 1.4.11 contrast requirements. So, for both radio buttons and checkboxes in a read-only state state (in both dark and light themes), the circle or square should meet a contrast ratio of at least 3:1.

**Comment by origami-z** [comment]
By reading the threads above, I don't think anyone has made any design changes yet.

**Comment by mark-tate** [comment]
Lungo Goal: @bhoppers2008 to define the Lungo goal

**Comment by mark-tate** [comment]
Macc Goal: solution ready for dev by EOS

**Comment by mark-tate** [comment]
red eye: @bhoppers2008 this is a defect, can we prioritise for red eye, maybe another designer can help get things going ?

Are these the only known contrast issues or should we be looking at them altogether?

@dplsek @jake-costa are going to help us get this into a known state

**Comment by mark-tate** [comment]
Red Eye Goal: complete proposal by EOS

**Comment by dplsek** [comment]
New design options [here](https://www.figma.com/design/ChsbbO7pLomT4F5H6tQyLP/Salt-(Next)-Components-%26-Patterns?m=auto&node-id=53243-4868&t=V2KATexHFZS6kdqj-1). Need to review with @bhoppers2008 @pseys @jake-costa

**Comment by bhoppers2008** [comment]
Design approach agreed. Implementation needs to be defined.

**Comment by bhoppers2008** [comment]
@bhoppers2008 to setup time to align.

**Comment by mark-tate** [comment]
Ginger Goal: complete by EOS
